pkg/pool/pool.go

```diff
@@ -103,7 +103,7 @@ func (p *Pool) recheck(ctx context.Context) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
-	for i := 0; i < len(p.endpoints); i++ {
+	for i := range len(p.endpoints) {
```

Comment:
Can be `for i:= range p.endpoints`

Reply:
I think about it, but why is it initially written this way?

Reply:
We can ask @smallhive, he's the author of d146024d.

Reply:
I saw it that way at that moment